server.py

```diff
@@ -209,44 +230,161 @@ def colorNewPlayer():
     return((randint(1,255),randint(1,255),randint(1,255)))
 
 
-
-# ----------------------- Handler -----------------------
-
-class MyTCPHandler(socketserver.BaseRequestHandler):
-    """
-    The request handler class for our server.
-
-    It is instantiated once per connection to the server, and must
-    override the handle() method to implement communication to the
-    client.
-    """
-
-    def handle(self):
-        # self.request is the TCP socket connected to the client
-        self.data = self.request.recv(1024).strip()
-        
-        in_ip = self.client_address[0]
-        
-        print("{} wrote:".format(in_ip))
-        in_data = str(self.data,'utf-16')
-        print(in_data)
+# ----------------------- Threads -----------------------
+def manage_server():
+    global STOP
+    global LISTENING
+    global MANAGING
+    
+    global MAINSOCKET
+    
+    while not STOP:
+        command = input()
+        match command:
+            case "stop":
+                STOP = True
+                print("STOP = ", STOP)
+                MAINSOCKET.shutdown(SHUT_RDWR)
+                MAINSOCKET.close()
+                
+                print("Socket server closed !")
+                
+                for (username,(sock,addr)) in dicoSocket:
+                    sock.shutdown(SHUT_RDWR)
+                    sock.close()
+                
+                print("Client sockets closed !")
+            case "deaf":
+                LISTENING = False
```

Comment:
we need to add " .shutdown(...)"

Reply:
but maybe we can do that later

Reply:
Yes we will have to find how to do it !
I quickly tried to add a shutdown call here but it did not go very well during my tests...
There must be another way to do so but I don't know yet how to do it.

Though we will have to find a solution as we must be able to have a maximum number of players and that the easiest way to do so would be to stop listening from new connections I think.

Reply:
We could eventually use a
```
if LISTENING:
    out = process_request(...)
```
after the `MAINSOCKET.accept()`

Reply:
We should definitely look closer at this : https://stackoverflow.com/questions/10090236/how-to-stop-a-python-socket-accept-call
I'll do it another day.